ast-nodes/variable-declaration.dart

```diff
@@ -63,6 +63,8 @@ class VariableDeclaration extends Declaration {
   }
 
   void checkSemantics() {
-    // TODO: implement
+    this.type.checkSemantics();
+    this.value.checkSemantics();
+    ensureType(this.value, this.type);
```

Comment:
`ensureType` produces no meaningful side effect, it returns a value. Also, `this.value` and `this.type` might both be `null` and `ensureType` will crash if you pass null as an argument.